File: go/sieve/sieve.go
package sieve

// Sieve returns all prime numbers up to (and including) limit.
func Sieve(limit int) (primes []int) {
	m := map[int]bool{}
	num := 2
	for num <= limit {
		if _, ok := m[num]; !ok {
			primes = append(primes, num)
			for i := num; i <= limit; i += num {
				m[i] = true
			}
		}
		num++
	}
	return
}

File: python/sieve/sieve.py
def sieve(limit):
  numbers = [0] * (limit+1) 
  primes = []

  n = 2
  while n <= limit:
    if numbers[n] == 0:
      primes.append(n)
      for i in range(n, limit+1, n):
        numbers[i] = 1
    n += 1

  return primes